### PR TITLE
Sudoku-Solver

### PR DESCRIPTION
**Pull request by Meghag5**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)
yes

### Description

(Write your answer here.)
This is a backtracking problem which is used for solving sudoku.
In sudoku there should be no repetition of number int same row, same column and same grid which of 3*3 size.
eg.
Input -
9
5 3 0 0 7 0 0 0 0
6 0 0 1 9 5 0 0 0
0 9 8 0 0 0 0 6 0
8 0 0 0 6 0 0 0 3
4 0 0 8 0 3 0 0 1
7 0 0 0 2 0 0 0 6
0 6 0 0 0 0 2 8 0
0 0 0 4 1 9 0 0 5
0 0 0 0 8 0 0 7 9

Output-
5 3 4 6 7 8 9 1 2
6 7 2 1 9 5 3 4 8
1 9 8 3 4 2 5 6 7
8 5 9 7 6 1 4 2 3
4 2 6 8 5 3 7 9 1
7 1 3 9 2 4 8 5 6
9 6 1 5 3 7 2 8 4
2 8 7 4 1 9 6 3 5
3 4 5 2 8 6 1 7 9

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)
#4271

### CODE REVIEW

File: C-Plus-Plus/math/sudoku_solver.cpp [hunk 1]
```diff
@@ -0,0 +1,122 @@
+// this is a backtracking problem which is used for solving sudoku
+// in sudoku there should be no repetition of number int same row, same column and same grid which of 3*3 size
```
Comment (HarshCasper) [inline C-Plus-Plus/math/sudoku_solver.cpp]: Use Multi-Line Comments

File: C-Plus-Plus/math/sudoku_solver.cpp [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+       return sudoku_solver(mat,i+1,0,n);
+    }
+    
+    // if the number filled is not 0 then we will skip it move to next column
+    if(mat[i][j] != 0){
+       return sudoku_solver(mat,i,j+1,n);
+    }
+
+    //recursive case
+    //fill the current cell with possible options
```
Comment (HarshCasper) [inline C-Plus-Plus/math/sudoku_solver.cpp]: Use Multi-Line Comments

File: C-Plus-Plus/math/sudoku_solver.cpp [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+            for(int j=0;j<n;j++){
+                cout<<mat[i][j]<<" ";
+            }
+            cout<<endl;
+        }
+        return true;
+    }
+
+    // if all the columns of a particular row is filled then we will move to next row
+    // and will start from the 0th column
```
Comment (HarshCasper) [inline C-Plus-Plus/math/sudoku_solver.cpp]: Use Multi-Line Comments

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by Meghag5** [comment]
@HarshCasper pls review all the changes required are made

**Pull request merged**